Makefile

```diff
@@ -9,6 +9,8 @@ IMG_REPO = $(shell echo $(IMG) | cut -d: -f1)
 IMG_TAG = $(shell echo $(IMG) | cut -d: -f2)
 # ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
 ENVTEST_K8S_VERSION = 1.32.0
+# Default image repo for kcm values during build: image.repository
+KCM_CHART_IMAGE_REPO ?= ghcr.io/k0rdent/kcm/controller
```

Comment:
Since this Makefile is quite busy already let's not introduce new variable, but reuse the `IMG_REPO`
I propose to not modify `set-kcm-version` target, but add new target `set-kcm-repo` with:
```makefile
$(YQ) eval '.image.repository = "$(IMG_REPO)"' -i $(PROVIDER_TEMPLATES_DIR)/kcm/values.yaml
```
and call it separately from the workflow along with `IMG` (yes the one with a full image name) env variable populated.

The rationale here is that we use `set-kcm-version` as a part of release process and it will make it little bit more confusing, since all `.image` values literally defined in the kcm chart will be overwritten.

So ultimately: I consider usage of `IMG` variable optional (since it's not ideal as well), but separate target is a must.